README.md

```diff
@@ -5,6 +5,65 @@ This repository is aimed to be added as git submodule into particular
 containers' source repositories.  By default, the path to submodule should be
 named 'common'.
 
+Usage
+-----
+
+This section explains the usage of the shared scripts in this repository when
+it is used as a submodule in a container source repository.
+
+Once you have the repository set as a submodule include `common.mk` into your
+root Makefile in order to access the default rules used to call shared scripts.
+
+**Default rules:**
+
+`make` or `make build`  
+This rule will build an image without tagging it with any tags after it is built.
+`make build` will also expect a `README.md` so that it can transfrom it into
+a man page that gets added to the image so make sure it is available and that
+you have the `go-md2man` tool installed on your host.
+
+`make tag`  
+Use this rule if you want to tag an image after it is built. It will be tagged with
+two tags - name:latest and name:version.
+Depends on `build`
+
+`make test` or `make check`  
+This rule will run the testsuite scripts contained in the container source repositories.
+It expects the test to be available at `$gitroot/$version/test/run`
+Depends on `tag` as some tests might need to have the images tagged (s2i).
+
+`make test-openshift`  
+Similar to `make test` but runs testsuite for Openshift, expected to be found at
+`$gitroot/$version/test/run-openshift`
+
+`make clean`  
+Best-effort to remove the last set of images that have been built using the scripts.
+
+**There are additional variables that you can use that the default rules are prepared to
+work with:**
+
+`VERSIONS`  
+Names of the directories in which the Dockerfiles are contained. Needs to be defined in your
+Dockerfile for the scripts to know which versions to build.
+
+`OS`  
+OS version you want to build the images for. Currently the scripts are able to build for
+centos (default), centos6, rhel7 and fedora.
+
+`SKIP_SQUASH`  
+When set to 1 the build script will skip the squash phase of the build.
+
+`CUSTOM_REPO`  
+Set this variable to the path to your local .repo files you want top have available inside
+the image while building. Useful for building rhel-based images on an unsubscribed box.
```

Comment:
Dunno, this might or might not be implemented in the image itself...  Something like
best practice, but it might not work without additional work in the depending project.

Reply:
This actually does not need any special modifications in the image, apart from not touching the mounted .repo files (which was the change needed in s2i-core). But I will make a note of it in the text